app/components/shared/Table/Table.tsx

```diff
@@ -146,17 +149,40 @@ export const Table = <T extends IdObj>({
             </thead>
             <tbody>
               {paginatedData.length > 0 ? (
-                paginatedData.map((item) => (
-                  <tr key={item.id}>
-                    {Object.entries(removeIdFromObject(item)).map(
-                      ([key, value]: TableDataArray) => (
-                        <td key={key}>
-                          {typeof value === "object" ? value.element : value}
-                        </td>
-                      ),
-                    )}
-                  </tr>
-                ))
+                paginatedData.map((item) => {
+                  return (
+                    <>
+                      {rowAsLink ? (
+                        <tr key={item.id}>
+                          <Link
+                            style={{
+                              display: "contents",
+                            }}
+                            to={`${item.id}`}
+                          >
+                            {Object.entries(removeIdFromObject(item)).map(
```

Comment:
I've got a few suggestions here... 

1- We can replace removeIdFromObject with  const `{id, ...rest} = item`, so we can destruct id and use it directly and access the rest of the data where needed and maybe create a constant `const tableData = Object.entries(data)` that can be used later instead of duplicating the same line twice. 

2- Table data cell rendering is duplicated twice and I think we can move it into one component (function).  So for example the code can be something like this:

```
const renderTableCell = ([key, value]) => (
    <td key={key}>
      {typeof value === "object" ? value.element : value}
    </td>
  ); 
```

and 

```
    <tr key={id}>
      {rowAsLink ? (
        <Link to={`${id}`} style={{ display: "contents" }}>
          {tableData.map(renderTableCell)}
        </Link>
      ) : (
        tableData.map(renderTableCell)
      )}
    </tr>
```

Reply:
Truth be told I don't think doing these things are worth it, this component is terrible and should be redone from 0. I mostly just copied what was already there to fit the new requirement, but I think the real solution is to make a more flexible component and make variants from there. 

If we think doing these small adjustments are worth it then I can do them, but I think this component is a lost cause.

Reply:
I totally agree, but I personally think the changes are small and would significantly improve the readability of the component and the ability to maintain and easily re-write it in the future.